### PR TITLE
Add support of google-test

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

File: gtest/CMakeLists.txt.in [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
+
+project(gtest NONE)
+
+include(ExternalProject)
+ExternalProject_Add(gtest
+  GIT_REPOSITORY    https://github.com/google/googletest.git
+  GIT_TAG           master
```
Comment (gergondet) [inline gtest/CMakeLists.txt.in]: Maybe we'd want to allow to specify a version for gtest?

I know they plan to follow the "live at head" philosophy but that might be a burden on the maintainer of jrl-cmakemodules-enabled project that simply wants its tests to keep working.

File: gtest.cmake [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set(CURRENT_FILE_PATH ${CMAKE_CURRENT_LIST_DIR} CACHE INTERNAL "")
```
[the remark this answers is not in the document]
Comment (jcarpent) [reply gtest.cmake]: This line allows defining the variable when loading the file, and during the evaluation of the macro.

File: gtest/CMakeLists.txt.in (hunk 1 above)
Comment (jcarpent) [reply gtest/CMakeLists.txt.in]: I just added an optional argument to ADD_GTEST_SUITE to account for a precise tag.

**Comment by jmirabel** [comment]
It might be worth adding the following to `.docs/pages/other.rst`:
```
.. cmake-module:: ../../gtest.cmake
```
Either in section *unit testing* or in a new section.

**Comment by jcarpent** [comment]
I just added this last requirement on top of the current master branch.

**Comment by jmirabel** [comment]
https://github.com/jrl-umi3218/jrl-cmakemodules/commit/a014e75ee10e80061cfbc646f207276e42dac305#r30720925

**Pull request merged**